src/Analyzers/Core/Analyzers/AbstractBuiltInCodeStyleDiagnosticAnalyzer.cs

```diff
@@ -35,12 +36,13 @@ internal abstract partial class AbstractBuiltInCodeStyleDiagnosticAnalyzer
         /// <param name="configurable">Flag indicating if the reported diagnostics are configurable by the end users</param>
         protected AbstractBuiltInCodeStyleDiagnosticAnalyzer(
```

Comment:
Are we sure that all code style analyzers inherits this abstract class (i.e. None of them inherits `DiagnosticAnalyzer` directly)?

Consider finding a good way to enforce that in the code base. (i.e. new analyzers don't miss this)

Reply:
> Are we sure that all code style analyzers inherits this abstract class (i.e. None of them inherits DiagnosticAnalyzer directly)?

Nope, instead we have some analyzers that actually don't, and these analyzers also end up missing linking their doc link to the analyzer diagnostics. I thought we had some issue to track fixing this up. If not, feel free to file one or better open a PR for it.

> Consider (In a follow-up PR) finding a good way to enforce that in the code base. (i.e. new analyzers don't miss this)

One way to enforce this could be to ban usage of DiagnosticDescriptor constructor in the CodeStyle folder by adding its doc comment ID to https://github.com/dotnet/roslyn/blob/master/src/CodeStyle/BannedSymbols.txt. We can add a suppression for this banned symbol usage in `AbstractBuiltInCodeStyleDiagnosticAnalyzer` and `AbstractCodeQualityDiagnosticAnalyzer` types, but get an analyzer diagnostic if any other code directly inherits from `DiagnosticAnalyzer` type and instantiates DiagnosticDescriptor.

Reply:
@Youssef1313 One example: https://github.com/dotnet/roslyn/blob/64769f8b6c13abd7e931c0cf7605368da83d74c5/src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs#L26-L27

This is turn causes IDE0005 diagnostics to not have an associated help link, falling back to bing search instead of https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0005: 
![image](https://user-images.githubusercontent.com/10605811/102384252-57b8ec00-3f81-11eb-9b4a-3bba9b63ab84.png)

Let me file an issue and tag you, in case you are interested in fixing this up

Reply:
https://github.com/dotnet/roslyn/issues/50011